queue_services/payment-reconciliations/src/reconciliations/ejv_reconciliations.py

```diff
@@ -0,0 +1,174 @@
+        elif is_jv_header:
+            journal_name: str = line[7:17]  # {ministry}{ejv_header_model.id:0>8}
+            ejv_header_model_id = int(journal_name[2:])
+            ejv_header: EjvHeaderModel = EjvHeaderModel.find_by_id(ejv_header_model_id)
+            ejv_header_return_code = line[271:275]
+            ejv_header.disbursement_status_code = _get_disbursement_status(ejv_header_return_code)
+            ejv_header_error_message = line[275:425]
+            ejv_header.message = ejv_header_error_message
+        elif is_jv_detail:
+            journal_name: str = line[7:17]  # {ministry}{ejv_header_model.id:0>8}
+            ejv_header_model_id = int(journal_name[2:])
+            invoice_id = int(line[205:315])
+            invoice: InvoiceModel = InvoiceModel.find_by_id(invoice_id)
+            invoice_link: EjvInvoiceLinkModel = db.session.query(EjvInvoiceLinkModel).filter(
+                EjvInvoiceLinkModel.ejv_header_id == ejv_header_model_id).filter(
+                EjvInvoiceLinkModel.invoice_id == invoice_id).one_or_none()
+
+            invoice_return_code = line[315:319]
+            invoice_return_message = line[319:469]
+            invoice_link.disbursement_status_code = _get_disbursement_status(invoice_return_code)
+            invoice_link.message = invoice_return_message
+            invoice.disbursement_status_code = _get_disbursement_status(invoice_return_code)
+            # If the JV process failed, then mark the GL code against the invoice to be stopped
+            # for further JV process for the credit GL.
+            is_credit: bool = line[104:105] == 'C'
+            if is_credit:
+                line_items: List[PaymentLineItemModel] = invoice.payment_line_items
+                for line_item in line_items:
+                    # Line debit distribution
+                    debit_distribution: DistributionCodeModel = DistributionCodeModel \
+                        .find_by_id(line_item.fee_distribution_id)
+                    credit_distribution: DistributionCodeModel = DistributionCodeModel \
+                        .find_by_id(debit_distribution.disbursement_distribution_code_id)
+                    credit_distribution.stop_ejv = True
+
+    db.session.commit()
+
+
+def _get_disbursement_status(return_code: str) -> str:
+    """Return disbursement status from return code."""
+    if return_code == '0000':
+        return DisbursementStatus.COMPLETED.value
+    return DisbursementStatus.ERRORED.value
+
+
+async def _publish_mailer_events(file_name: str, minio_location: str):
+    """Publish payment message to the mailer queue."""
+    # Publish message to the Queue, saying account has been created. Using the event spec.
+    queue_data = {
+        'fileName': file_name,
+        'minioLocation': minio_location
+    }
+    payload = {
+        'specversion': '1.x-wip',
+        'type': 'bc.registry.payment.ejvFailed',
+        'source': 'https://api.pay.bcregistry.gov.bc.ca/v1/accounts/',
```

Comment:
Config value for this one? In case we are changing domain names.

Reply:
Yea, we are not using the source.. Just keeping there to match with event spec. Valid point, we can move it config